### PR TITLE
Zkapps nonce integration test

### PR DESCRIPTION
**Pull request by psteckler**
Same test as in #12264, but atop `updated-zkapps-txn-pool-model-merged`.

Author: @MartinMinkov

### CODE REVIEW

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Review by lk86 — APPROVED** [review]
Ceremonial infra stamp, though i doubt i'm best suited to review

**Comment by psteckler** [comment]
!ci-build-me

File: src/lib/zkapp_command_builder/zkapp_command_builder.ml [hunk 1]
```diff
@@ -10,27 +10,35 @@ let mk_forest ps :
...
+  let preconditions =
+    Option.value preconditions
+      ~default:
+        Account_update.Preconditions.
+          { network = Zkapp_precondition.Protocol_state.accept
+          ; account = Account_update.Account_precondition.Accept
+          ; valid_while = Ignore
+          }
+  in
+  let increment_nonce = Option.value increment_nonce ~default:false in
```
Comment (nholland94) [inline src/lib/zkapp_command_builder/zkapp_command_builder.ml:26]: NIT: you could use the `let f ?(x = default)` syntax here to make this a bit cleaner.

File: src/app/test_executive/zkapps_nonce_test.ml [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+    in
+    let%bind () =
+      section
+        "Send a zkapp command that has its nonce properly incremented after \
+         the fish1 transaction"
+        (send_zkapp ~logger node valid_zkapp_cmd_from_fish1)
+    in
+    let%bind () =
+      section
+        "Wait for fish1 zkapp command with invalid nonce to be rejected by \
```
Comment (nholland94) [inline src/app/test_executive/zkapps_nonce_test.ml:226]: NIT: "rejected by frontier" is odd wording here; the txn is accepted into the frontier, but with a failed status.

**Review by nholland94 — APPROVED** [review]
The test LGTM!

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Pull request merged**